src/modm/platform/core/cortex/module.lb

```diff
@@ -200,17 +200,20 @@ def prepare(module, options):
 
     memories = listify(options[":target"].get_driver("core")["memory"])
 
-    # Cortex-M0 does not have remappable vector table, so it will remain in Flash
-    if not options[":target"].has_driver("core:cortex-m0*"):
-        default_location = "rom"
-        if any((m["name"] == "ccm" and "x" in m["access"]) or m["name"] == "dtcm" for m in memories):
-            default_location = "ram"
-        module.add_option(
-            EnumerationOption(
-                name="vector_table_location",
-                description=FileReader("option/vector_table_location.md"),
-                enumeration=["rom", "ram"],
-                default=default_location))
+    # Cortex-M0 does not have a remappable vector table but for most devices the vector table
+    # can be placed in ram nevertheless. A platform specific remap method is used
+    # which maps the start of ram to address 0x0 where the vector table resides.
+    # The remap has to be implemented in the platform-specific core module.
+    # Many Cortex-M0+ have a remappable vector table, at least all devices supported in modm do.
+    default_location = "rom"
+    if any((m["name"] == "ccm" and "x" in m["access"]) or m["name"] == "dtcm" for m in memories):
+        default_location = "ram"
+    module.add_option(
+        EnumerationOption(
+            name="vector_table_location",
+            description=FileReader("option/vector_table_location.md"),
+            enumeration=["rom", "ram"],
+            default=default_location))
```

Comment:
No, let this be. There should be a separate, vendor-specific `:platform:core:vector_table_location`. This option is for VTOR specific behavior, which relocates the NVIC pointer, rather than remapping the entire application. It's not the same thing, and has different side-effects.

Reply:
I had this first, but ran into issues with lbuild. I would like to reuse all the code in the generic cortex-m module that copies the vector table to ram, allocates the memory for it in ram, etc. What would you suggest to get both? Basically I need to do the same things as if `vector_table_location` was set to ram in the cortex-m module but do something different on the line setting `SCB->VTOR`.

Reply:
Hm, you could change this line here to OR itself properly with `env.get(":platform:core:vector_table_location", "rom")`. It would still be vendor neutral, since `modm:platform:core` is the same name for STM32 and SAM.

The actual VTOR setting is guarded by `%% if "m0" not in core`, so shouldn't be affected.

Reply:
That works nicely :+1: . Pretty weird I got an lbuild error accessing the :platform:core module option from the cortex-m module when I tried first. I don't know what I did wrong there …

Reply:
lbuild complains when you access the options via `env["module:option"]` without depending on the module, because that will fail if the module is not included. However, if you provide a default `env.get("module:option", default)` then it won't fail and therefore won't complain.